themes/default/content/templates/kubernetes/webapp/index.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Web Application on Kubernetes
+layout: template
+
+meta_desc: The Kubernetes Application template makes it easy to deploy a web application to a Kubernetes cluster with Pulumi.
+
+meta_image: meta.png
+
+card_desc: Deploy a web application to a Kubernetes cluster with Pulumi.
+
+template:
+  prefix: webapp-kubernetes
+  dirname: my-k8s-app
+  languages:
+    - typescript
+    - python
+    - go
+    - csharp
+    - yaml
+
+cloud:
+  name: Kubernetes
+  slug: kubernetes
+
+---
+
+The Kubernetes Application template creates an infrastructure as code project in your favorite language that deploys a web application to an existing cluster with Pulumi. It uses Pulumi's [Native Kubernetes Provider](/registry/packages/kubernetes) to create a new namespace in an existing Kubernetes cluster and a new deployment to host the application. It also exposes the application as a service that's reachable from outside your cluster. The template generates a complete infrastructure as code program, including sample application code, to give you a working project out of the box that you can customize easily and extend to suit your needs.
+
+![An architecture diagram of the Pulumi Kubernetes Application template](./architecture.png)
+
+## Using this template
+
+To use this template to deploy a web application to a Kubernetes cluster, make sure you've already provisioned a [Kubernetes cluster](/templates/kubernetes), [installed Pulumi](/docs/get-started/install) and [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/), and [configured your kubeconfig file](/registry/packages/kubernetes/installation-configuration#setup). Then create a new [project](/docs/intro/concepts/project) using the template in your language of choice:
+
+{{< templates/pulumi-new >}}
+
+Follow the prompts to complete the new-project wizard. When it's done, you'll have a complete Pulumi project that's ready to deploy and configured with the most common settings. Feel free to inspect the code in {{< langfile >}} for a closer look.
+
+## Deploying the project
+
+The template requires no additional configuration. Once the new project is created, you can deploy it immediately with [`pulumi up`](/docs/reference/cli/pulumi_up):
+
+```bash
+$ pulumi up
+```
+
+When the deployment completes, Pulumi exports the following [stack output](/docs/intro/concepts/stack#outputs) values:
+
+deploymentName
+: The name of your new Kubernetes Deployment.
+
+serviceName
+: The name of your new Kubernetes Service.
+
+Output values like these are useful in many ways, most commonly as inputs for other stacks or related cloud resources. The computed `serviceName`, for example, can be used from the command line to open the newly deployed web application in your favorite web browser:
```

Comment:
Unfortunately, you can't use the computed `serviceName` output in this way. Although it is exposed as a Service, it's not consumable using that URL.